contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -103,9 +102,9 @@ contract TrueCreditAgency is UpgradeableClaimable {
 
     function _creditScoreAdjustmentRate(uint8 score) internal view returns (uint256) {
         if (score == 0) {
-            return type(uint256).max;
```

Comment:
The interest rate cap is owner-settable, so we should extract this into a state variable. Not the core part of this PR, so we can add a TODO to handle it later.

And to make sure we're on the same page: we want to cap the _overall_ interest rate at 500%. It shouldn't matter how the intermediate calculations propagate values that are over the cap, but I think the code would be cleaner if we made a saturatingAdd() math library function that capped the individual `+`s in `baseRate + riskPremium + creditAdjustment + utilizationAdjustment` at type(uint256).max, and then applied the 500% cap only once at the end.

Reply:
I think it would be just as clean if we had this maxRateCap and neither component of the rate calculation would not surpass it. This will remove the necessity to take care about overflows when adding anything to INT_MAX

Reply:
Yeah that makes sense. It's worth it to avoid having to do overflow checks for arithmetic near INT_MAX.